x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/90_semantic_text_source_filtering.yml

```diff
@@ -0,0 +1,125 @@
+        index: test-index
+        body:
+          mappings:
+            properties:
+              sparse_field:
+                type: semantic_text
+                inference_id: sparse-inference-id
+
+  - do:
+      index:
+        index: test-index
+        id: doc_1
+        body:
+          sparse_field: ["inference test", "another inference test"]
+        refresh: true
+
+---
+"Exclude embeddings from _source by default":
+
+  - requires:
+      cluster_features: "gte_v8.16.0"
+      reason: _source filtering for Semantic text added in 8.16.0
+
+  - do:
+      headers:
+        # Force JSON content type so that we use a parser that interprets the floating-point score as a double
+        Content-Type: application/json
+      search:
+        index: test-index
+        body:
+          query:
+            semantic:
+              field: "sparse_field"
+              query: "inference test"
+
+  - not_exists: _source.sparse_field.inference.chunks.0.embeddings
+  - not_exists: _source.sparse_field.inference.chunks.1.embeddings
+
+---
+"Exclude embeddings from _source":
+
+  - requires:
+      cluster_features: "gte_v8.16.0"
+      reason: _source filtering for Semantic text added in 8.16.0
+
+  - do:
+      headers:
+        # Force JSON content type so that we use a parser that interprets the floating-point score as a double
+        Content-Type: application/json
+      search:
+        index: test-index
+        body:
+          _source:
+            include_vectors: false
+          query:
+            semantic:
+              field: "sparse_field"
+              query: "inference test"
+
+  - not_exists: _source.sparse_field.inference.chunks.0.embeddings
+  - not_exists: _source.sparse_field.inference.chunks.1.embeddings
+
+---
+"Include embedding in _source":
+
+  - requires:
+      cluster_features: "gte_v8.16.0"
+      reason: _source filtering for Semantic text added in 8.16.0
+
+  - do:
+      headers:
+        # Force JSON content type so that we use a parser that interprets the floating-point score as a double
+        Content-Type: application/json
+      search:
+        index: test-index
+        body:
+          _source:
+            include_vectors: true
+          query:
+            semantic:
+              field: "sparse_field"
+              query: "inference test"
+
+  - exists: _source.sparse_field.inference.chunks.0.embeddings
+  - exists: _source.sparse_field.inference.chunks.1.embeddings
+
+---
+"Include embeddings in _source by default":
+
+  - do:
+      headers:
+        # Force JSON content type so that we use a parser that interprets the floating-point score as a double
+        Content-Type: application/json
+      search:
+        index: test-index
+        body:
+          query:
+            semantic:
+              field: "sparse_field"
+              query: "inference test"
+
+  - exists: _source.sparse_field.inference.chunks.0.embeddings
+  - exists: _source.sparse_field.inference.chunks.1.embeddings
```

Comment:
Once again, this test can't pass. Any reason why you are adding it?

Reply:
The idea was to use the `8.15` cluster and verify backward compatibility.

Reply:
Ah, you'll need to restrict the ES versions this test runs on then. Would be a good idea to indicate the BwC purpose in the test name as well.

Reply:
I am selecting the `8.15` cluster for this test suite defined [here](https://github.com/elastic/elasticsearch/pull/112319/files/e146cf57104b42e7fda7a18dab91c01fef6ee8d3#diff-6d8a5c9de0420df64db6b61960a3b1e9bed8ef346c31a3abaa2abb5ee26530fbR3) and not overriding the version. Should not that be enough? Is there any other way to set or restrict the ES version?

Reply:
You need to restrict this test so that it only runs on 8.15. As you've defined at the beginning of this file, this entire test suite will run on 8.15 and newer. You need to restrict this particular test so that it runs only on 8.15.